pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -769,59 +791,64 @@ func (pl *dynamicResources) PreScore(ctx context.Context, cs *framework.CycleSta
 	}
 
 	logger := klog.FromContext(ctx)
```

Comment:
should we call `defer preScored()` here?

If you only want it to be called under certain circumstances, perhaps the name is wrong.

Reply:
You are right, setting `state.preScored = true` on all exits via `defer` is better. Changed.